jerry-core/mem/mem-heap.c

```diff
@@ -243,46 +186,48 @@ void *mem_heap_alloc_block_internal (const size_t size)
   const size_t required_size = ((size + MEM_ALIGNMENT - 1) / MEM_ALIGNMENT) * MEM_ALIGNMENT;
   mem_heap_free_t *data_space_p = NULL;
 
-  VALGRIND_DEFINED_SPACE (&mem_heap.first, sizeof (mem_heap_free_t));
+  mem_heap_t *mem_heap = jerry_port_get_heap ();
```

Comment:
I am worried about the performance cost of this change. Especially because the base address is required for all heap pointer compression and decompression. So I think we should keep the base address and other properties in global variables.

Perhaps the following API would have a smaller overhead:

    struct { uint8_t area[MEM_HEAP_AREA_SIZE] /* forced alignment by posix memalign or something */ } heap_area;

    // Jerry calls this only once. Returned heap is stored in a global variable
    heap_area* jerry_port_alloc_heap(size_t size, size_t alignment); 
    // Jerry calls this only once. Free heap.
    void jerry_port_free_heap(heap_area* area); 

This will also have a performance overhead, we should measure it first before we make a decision.